### ISSUE TITLE
Is there a release date for Typescript support?

### ISSUE THREAD

**Issue by cody-ta** [getbrevo/brevo-node#1]
My current project used Typescript, I wasn't able to migrate from SIB to Brevo.

<img width="324" alt="CleanShot 2023-06-15 at 16 13 51@2x" src="https://github.com/getbrevo/brevo-node/assets/8092790/55e48165-9b48-4994-ae47-6463ae477c96">

**Comment by shubhamUpadhyayInBlue** [comment]
Hi @cody-ta 
We are still trying to find a way to provide TS support for the same. It is taking us some time to figure that out since our SDKs are auto-generated using swagger-codegen@2.
Thanks.

**Comment by reynolga** [comment]
Also looking for typescript support as I'm moving my website from js to typescript at the moment.

**Comment by shubhamUpadhyayInBlue** [comment]
Hello
I have raised the request and will update you as soon as I have any updates. Thanks.

**Comment by matthiasbruns** [comment]
Until solved:

`swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o ./lib/email/brevo.d.ts`

**Comment by NarHakobyan** [comment]
> Until solved:
> 
> `swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o ./lib/email/brevo`

Better then lib 😄

**Comment by AlexZeitler** [comment]
@matthiasbruns How do you set the API key?

**Comment by NarHakobyan** [comment]
> @matthiasbruns How do you set the API key?

```typescript
new Brevo({
          baseApiParams: {
            headers: {
              'api-key': yourKeyhere,
            },
          },
        })
````

**Comment by AlexZeitler** [comment]
@NarHakobyan Thanks!

**Comment by shubhamUpadhyayInBlue** [comment]
The latest swagger file is here: `https://api.brevo.com/v3/swagger_definition_v3.yml`

So, @matthiasbruns @NarHakobyan you can try using this latest swagger file instead.

**Comment by garysb** [comment]
Hi @shubhamUpadhyayInBlue, thanks for the reference but there is an error in the swagger file `https://api.brevo.com/v3/swagger_definition_v3.yml` at line 6656, you have an extra double quotation mark after delivered.

```
✨   swagger-typescript-api(12.0.4),nodejs(v18.15.0)
✨   try to read templates from directory "/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/templates/default"
✨   "api" template found in "/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/templates/default"
✨   "route types" template found in "/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/templates/default"
✨   try to get swagger by URL "https://api.brevo.com/v3/swagger_definition_v3.yml"
☄️    start generating your typescript api
SyntaxError: ';' expected. (6656:97)
  6654 |    * @example "request"
  6655 |    */
> 6656 |     event: "invalid_parameter" | "missing_parameter" | "hardBounce" | "softBounce" | "delivered"" | "spam" | "request" | "opened" | "click" | "invalid" | "deferred" | "blocked" | "unsubscribed" | "error" | "uniqueOpened" | "loadedByProxy" | "allEvents",
       |                                                                                                 ^
  6657 |   /**
  6658 |    * Webhook URL to receive CSV file link
  6659 |    * @example "https://brevo.com"
    at Ve (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/parser-typescript.js:1:15607)
    at vz (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/parser-typescript.js:280:5919)
    at Object.yz [as parse] (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/parser-typescript.js:280:6242)
    at Object.parse (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/index.js:7334:23)
    at coreFormat (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/index.js:8645:18)
    at formatWithCursor2 (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/index.js:8837:18)
    at /Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/index.js:37229:12
    at Object.format (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/node_modules/prettier/index.js:37243:12)
    at CodeFormatter.prettierFormat (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/src/code-formatter.js:39:21)
    at CodeFormatter.formatCode (/Users/garysb/Projects/Scaut/api/node_modules/swagger-typescript-api/src/code-formatter.js:47:19) {
  loc: { start: { line: 6656, column: 97 } },
```

**Comment by garysb** [comment]
Hi @shubhamUpadhyayInBlue, thanks for the new definition but there is an issue in the swagger file at line `17986`. Please remove the double quote.

```
   try to get swagger by URL "https://api.brevo.com/v3/swagger_definition_v3.yml"
☄️    start generating your typescript api
SyntaxError: ';' expected. (6656:97)
  6654 |    * @example "request"
  6655 |    */
> 6656 |     event: "invalid_parameter" | "missing_parameter" | "hardBounce" | "softBounce" | "delivered"" | "spam" | "request" | "opened" | "click" | "invalid" | "deferred" | "blocked" | "unsubscribed" | "error" | "uniqueOpened" | "loadedByProxy" | "allEvents",
       |                                                                                                 ^
  6657 |   /**
  6658 |    * Webhook URL to receive CSV file link
  6659 |    * @example "https://brevo.com"
```

**Comment by shubhamUpadhyayInBlue** [comment]
Hi @garysb 
I have fixed the swagger file. Please try and let me know if you face any issue.

**Comment by marentdev** [comment]
Take me sometimes to get it working (because din't know how swagger-typescript-api was working) but now everything is working and tested!

https://gist.github.com/marentdev/28d49bfee47704ac39fcd55cb4ac8b0e (Brevo Lib)

**Comment by higuitadiaz** [comment]
Hi @shubhamUpadhyayInBlue 
Do you have defined an estimated time for an official documentation of the library under TS?

Developer:

If you do not know how to use the ```swagger-typescript-api```   library you can guide yourself with the following steps:

```javascript
/** npx swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o [PATH-OUTPUT] -n [TS-FILENAME (Sample: Brevo.ts)]
**/

// SAMPLE
// console
npx swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o /src/utils/email -n Brevo.ts
...

// ts-file.ts
import { Brevo } from './utils/email/Brevo';

...

const brevo = new Brevo({ baseApiParams: { headers: { 'api-key': 'YOUR-KEY' } })

```

**Comment by kracas** [comment]
> Hi @shubhamUpadhyayInBlue Do you have defined an estimated time for an official documentation of the library under TS?
> 
> Developer:
> 
> If you do not know how to use the `swagger-typescript-api` library you can guide yourself with the following steps:
> 
> ```js
> /** npx swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o [PATH-OUTPUT] -n [TS-FILENAME (Sample: Brevo.ts)]
> **/
> 
> // SAMPLE
> // console
> npx swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o /src/utils/email -n Brevo.ts
> ...
> 
> // ts-file.ts
> import { Brevo } from './utils/email/Brevo';
> 
> ...
> 
> const brevo = new Brevo({ baseApiParams: { headers: { 'api-key': 'YOUR-KEY' } })
> ```
Hi @higuitadiaz, 
Thanks for the explanation. 
Just a few comments, the swagger definition in your guide is outdated. The link to the current swagger definition is https://api.brevo.com/v3/swagger_definition_v3.yml (notice the second v3 at the end). 
Also swagger-typescript-api does not name the client Brevo by default. The name for the client is Api, so you should either change the name to Brevo in the Brevo.ts file or use it like this:
```js
// ts-file.ts
import { Api } from './utils/email/Brevo';

...

const brevo = new Api({ baseApiParams: { headers: { 'api-key': 'YOUR-KEY' } })
```

**Comment by SMKH-PRO** [comment]
Really need ts support before we can migrate to brevo

**Comment by shorthander** [comment]
I also need ts support before I can move to brevo... 
@shubhamUpadhyayInBlue have you thought about moving to the openapi generator? In openapi generator, which should be compatible to your swagger definition file, you should be able to generate typescript code for several libs like fetch or axios.
https://openapi-generator.tech/docs/generators

**Comment by cwierzbicki00** [comment]
Any update here @shubhamUpadhyayInBlue  ? 

We do not want to continue building with deprecated lib, but the lack of Typescript support here is blocking us.  It has been several months since you've raised this issue, so I would hope some progress has been made. Could you provide us with an estimate?

**Comment by shubhamUpadhyayInBlue** [comment]
Hi @cwierzbicki00 
It is under discussion and we will be able to give feedback soon. I will make sure your message reaches the concerned team.
Thanks.

**Comment by iampapagray** [comment]
> > Hi @shubhamUpadhyayInBlue Do you have defined an estimated time for an official documentation of the library under TS?
> > Developer:
> > If you do not know how to use the `swagger-typescript-api` library you can guide yourself with the following steps:
> > ```js
> > /** npx swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o [PATH-OUTPUT] -n [TS-FILENAME (Sample: Brevo.ts)]
> > **/
> > 
> > // SAMPLE
> > // console
> > npx swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o /src/utils/email -n Brevo.ts
> > ...
> > 
> > // ts-file.ts
> > import { Brevo } from './utils/email/Brevo';
> > 
> > ...
> > 
> > const brevo = new Brevo({ baseApiParams: { headers: { 'api-key': 'YOUR-KEY' } })
> > ```
> 
> Hi @higuitadiaz, Thanks for the explanation. Just a few comments, the swagger definition in your guide is outdated. The link to the current swagger definition is https://api.brevo.com/v3/swagger_definition_v3.yml (notice the second v3 at the end). Also swagger-typescript-api does not name the client "Brevo" by default. The name for the client is "Api", so you should either change the name to Brevo in the Brevo.ts file or use it like this:
> 
> ```js
> // ts-file.ts
> import { Api } from './utils/email/Brevo';
> 
> ...
> 
> const brevo = new Api({ baseApiParams: { headers: { 'api-key': 'YOUR-KEY' } })
> ```

Thanks for the Ts help.
Can anyone help with the the steps to send a simple transactional email with a template and a param? 
I cant seem to follow the instructions with this ts file.

**Comment by theofrgs** [comment]
Is there any update about the typescript release ?

**Comment by FredTreg** [comment]
```
$ npm remove @sendinblue/client
$ npm install @getbrevo/brevo
$ npm i --save-dev @types/getbrevo__brevo
npm ERR! code E404
```
=> Googling, finding this github issue, then 5 minutes later:

```
$ npm remove @getbrevo/brevo
$ npm install @sendinblue/client
```
:)

(I do hope they support TS soon though as the security audit of `sendinblue/client` is bad)

**Comment by vladislavbogomolov** [comment]
Sending transactional emails via axios/fetch is more comfortable

**Comment by dkrest1** [comment]
> Until solved:
> 
> `swagger-typescript-api -p https://api.brevo.com/v3/swagger_definition.yml -o ./lib/email/brevo`

how do I make use of this

**Comment by dkrest1** [comment]
> Is there any update about the typescript release ?

I am not sure because I have not seen one

**Comment by akouyate** [comment]
Up !

**Comment by mwolf1989** [comment]
Is there something new on this? Send In blue is gone now for a while and still missleading docs on the page, still no TS support.

**Comment by ryan-james-smith** [comment]
> Hi @cwierzbicki00 It is under discussion and we will be able to give feedback soon. I will make sure your message reaches the concerned team. Thanks.

If soon as in "we will be able to give feedback soon" is over two months, I guess I may need to look for a different company than brevo as Typescript support is non negotiable in our projects these days.
Would love an update with some good news very very soon in the hopes I don't have to rewrite what we are already using brevo for.

**Comment by shubhamUpadhyayInBlue** [comment]
Hello!
I have released a beta tag with typescript support. Here are the steps how you can try it out and share your valuable feedback.

To install it in your application:
```npm i @getbrevo/brevo@2.0.0-beta.4 --save```

Sample code to send the transactional email:

```
const SibApiV3Sdk = require('@getbrevo/brevo');
let apiInstance = new SibApiV3Sdk.TransactionalEmailsApi();
let apiKey = apiInstance.authentications['apiKey'];
apiKey.apiKey = 'YOUR API KEY';
let sendSmtpEmail = new SibApiV3Sdk.SendSmtpEmail(); 

sendSmtpEmail.subject = "My {{params.subject}}";
sendSmtpEmail.htmlContent = "<html><body><h1>This is my first transactional email {{params.parameter}}</h1></body></html>";
sendSmtpEmail.sender = {"name":"John Doe","email":"example@example.com"};
sendSmtpEmail.to = [{"email":"example@example.com","name":"Jane Doe"}];
sendSmtpEmail.cc = [{"email":"example2@example2.com","name":"Janice Doe"}];
sendSmtpEmail.bcc = [{"name":"John Doe","email":"example@example.com"}];
sendSmtpEmail.replyTo = {"email":"replyto@domain.com","name":"John Doe"};
sendSmtpEmail.headers = {"Some-Custom-Name":"unique-id-1234"};
sendSmtpEmail.params = {"parameter":"My param value","subject":"New Subject"};

apiInstance.sendTransacEmail(sendSmtpEmail).then(function(data) {
  console.log('API called successfully. Returned data: ' + JSON.stringify(data));

}, function(error) {
  console.error(error);
});

```

Hope it helps.
Thanks.

**Comment by mwolf1989** [comment]
Are there news vor the V2 release?

**Comment by zolzaya** [comment]
Is there any news about TypeScript?

**Comment by peterbbeu** [comment]
Quite annoying that I have to choose between typescript support, or a version with unpatchable vulnerabilities.

Come on, it's not that hard!

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ Server-Side Request Forgery in Request                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ request                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @getbrevo/brevo                                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @getbrevo/brevo > request                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1092972                     │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ tough-cookie Prototype Pollution vulnerability               │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tough-cookie                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.1.3                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @getbrevo/brevo                                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @getbrevo/brevo > request > tough-cookie                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1095102                     │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by alexbrazier** [comment]
I've created a PR https://github.com/getbrevo/brevo-node/pull/19 that hopefully solves this issue. Hopefully it gets accepted and merged, but in the meantime you can use [@groupcards/brevo](https://www.npmjs.com/package/@groupcards/brevo) which you can verify was published from [this action](https://github.com/squareda/brevo-node/actions/runs/8553009354).

Feel free to leave a comment on the PR if there are any issues with it or if it's working for you.

## Usage

`npm install @groupcards/brevo`

```ts
const brevo = new BrevoApi({
  headers: {
    "api-key": "YOUR_API_KEY",
  },
});

const account = await brevo.account.getAccount()
```

**Comment by jesus997** [comment]
I can't believe it's February 2025 and we still don't have full Typescript support.
